3-reveal/migrations/4-FI/1-Thailand-2019/deploy/events_jurisdictions_materialized_view.psql

```diff
@@ -0,0 +1,32 @@
+-- Deploy reveal_fi_thailand_2019:events_jurisdictions_materialized_view to pg
+-- requires: reveal_transaction_tables:events
+-- requires: reveal_transaction_tables:plans
+-- requires: reveal_transaction_tables:goals
+-- requires: reveal_transaction_tables:actions
+-- requires: utils:refresh_mat_views
+
+BEGIN;
+
+SET search_path TO :"schema",public;
+
+CREATE MATERIALIZED VIEW IF NOT EXISTS events_jurisdictions_materialized_view
+AS
+SELECT
+    events.id AS event_id,
+    events.details->'case_classification' AS case_classification,
+    events.details->'case_number' AS case_number,
+    events.details->'investigtion_date' AS date_of_diagnosis,
+    plans.identifier AS plan_id,
+    goals.goal_id,
+    events.location_id AS jurisdiction_id,
+    actions.code AS action_code
+FROM events
+LEFT JOIN plans
+    ON plans.fi_opensrpeventid = events.id
+LEFT JOIN goals
+    ON goals.plan_id = plans.identifier AND goals.goal_id = 'Case_Confirmation'
+LEFT JOIN actions
```

Comment:
If you join on actions you don't have to join on goals because actions have the goal_id that we need.

Reply:
@lincmba dont forget about this one